### PR TITLE
Decentral economy config

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

This creates `agoric-sdk/packages/vats/decentral-economy-config.json` which boots up a lot more of the economy than its predecessors.  It takes advantage of arguments to the `defaultCoreProposal` function.

I am requesting review and intend to merge, even though there are still some manual governance pieces to perform, namely:

- The oracle governance currently needs to execute in an ag-solo environment.  Instagoric should run it from the ag-solo to produce the oracleAdmin facet that can be used to push prices.  It would make sense to have the (single member) econ committee be just that privileged ag-solo instance.
- Once the oracle is sorted out, the economic committee will have to vote to raise the vault debt ceiling for that collateral.

### Testing Considerations

Tested with local-chain as well as sim-chain.

### CODE REVIEW

File: golang/cosmos/x/vbank/vbank.go
Comment (dckc) [inline golang/cosmos/x/vbank/vbank.go:147]: How did you discover this bug? Did some other part of this PR trip it?

File: packages/deploy-script-support/src/coreProposalBehavior.js
Comment (dckc) [inline packages/deploy-script-support/src/coreProposalBehavior.js]: Did you run into an example of interference? What was it?

Fixes should come with tests. Is writing a test so costly that it's not worthwhile in this case?

Comment (dckc) [reply packages/deploy-script-support/src/coreProposalBehavior.js]: Or perhaps the changes to `core-proposal.js` are tested (in the solo test-home.js?) and depend on this fix, so while we don't have a unit test for just this fix, we do run this code each ci?

File: packages/deploy-script-support/src/extract-proposal.js
Comment (dckc) [inline packages/deploy-script-support/src/extract-proposal.js:71]: features should have tests too... but again, I suppose the tests for the changes to `core-proposal.js` indirectly test this.

File: packages/run-protocol/scripts/add-collateral-core.js
Comment (dckc) [inline packages/run-protocol/scripts/add-collateral-core.js]: whoa... the scoping in default arguments is wild.

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (dckc) [inline packages/run-protocol/src/vaultFactory/addAssetToVault.js:25]: tangentially... is it a feature or a bug that this governance script now lives in the same directory as the smart contract? It was originally developed in hack-and-slash mode. Putting it here makes it more likely to be considered "in scope" for security reviews. I suppose PRs such as this one are polishing it in due course.

cc @arirubinstein @jessysaurusrex @turadg @dtribble @Chris-Hibbert

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (dckc) [inline packages/run-protocol/src/vaultFactory/addAssetToVault.js]: managing the scope of these option names seems a bit tricky. Do we document the things they need to not collide with anywhere?

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (dckc) [inline packages/run-protocol/src/vaultFactory/addAssetToVault.js]: add some prose docs for these?

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (dckc) [inline packages/run-protocol/src/vaultFactory/addAssetToVault.js]: yikes! how did that ever work before?!

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (dckc) [inline packages/run-protocol/src/vaultFactory/addAssetToVault.js:211]: So you have made your peace with this idiom after all?

**Review by dckc — COMMENTED** [review]
I don't think I found any critical problems, but I'd like to discuss the level of documentation and perhaps test coverage before I approve.

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (turadg) [reply packages/run-protocol/src/vaultFactory/addAssetToVault.js:25]: Agree with the principle that `run-protocol/src` contains logic for the contracts as they run, not how they're deployed. The value in keeping them separated is both for security review and for developers looking to this source for writing contracts and re-using these.

Thankfully this is used only for `scripts/add-collateral-core.js` so it would be easy as moving it over there. Longer term would it make sense to have another package (`@agoric/inter-protocol-vats`?) for the vat setup concern?

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (michaelfig) [reply packages/run-protocol/src/vaultFactory/addAssetToVault.js:211]: It's better than a string, even if it's still a little fragile.

File: golang/cosmos/x/vbank/vbank.go
Comment (michaelfig) [reply golang/cosmos/x/vbank/vbank.go:147]: Yeah, my crazy `@INTERCHAIN_DENOM@` template variable caused a Golang panic.  Drive-by fix for all the similar places.

File: packages/deploy-script-support/src/coreProposalBehavior.js
Comment (michaelfig) [reply packages/deploy-script-support/src/coreProposalBehavior.js]: I may have fixed the issue now by reserving the econ committee namesByAddress.  I'll try reverting this code to be sure.

File: packages/deploy-script-support/src/extract-proposal.js
Comment (michaelfig) [reply packages/deploy-script-support/src/extract-proposal.js:71]: I've been testing manually, but will automate if you have a suggestion how.

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (michaelfig) [reply packages/run-protocol/src/vaultFactory/addAssetToVault.js]: I've bundled them in `options.interchainAssetOptions` instead.  That seems to be the right kind of namespacing.

File: packages/run-protocol/src/vaultFactory/addAssetToVault.js
Comment (michaelfig) [reply packages/run-protocol/src/vaultFactory/addAssetToVault.js]: I created a separate type that should make this self-documenting.

File: packages/deploy-script-support/src/extract-proposal.js
Comment (dckc) [reply packages/deploy-script-support/src/extract-proposal.js:71]: I was thinking that `test-home.js` tested `core-proposal.js` already.

**Review by dckc — APPROVED** [review]
Ideally we'd have more automated / regression tests, but I reproduced your results, and `test-gov-collateral.js` exercises much of this, and the cost of more automated tests looks pretty high.

So let's go for it.

**Pull request merged**
